### PR TITLE
[8.0] Fix deployment of 8.0.x and add protections against deploying incorrect versions

### PR DESCRIPTION
**Pull request by chrisburr**
Partly deals with https://github.com/DIRACGrid/DIRAC/issues/6362 and adds some protection against deploying releases from integration or wrong versions from `rel-v*` branches.

I've tested this as carefully as I can locally but it's still likely to be buggy 😅 

It's only for the CI so not worthy of release notes.

### CODE REVIEW

File: .github/workflows/deployment.yml [hunk 1]
```diff
@@ -64,7 +64,7 @@ jobs:
           IFS=$'\n\t'
           if [[ "${{ github.repository }}" == "DIRACGrid/DIRAC" ]]; then
             if [[ "${{ github.event_name }}" == "workflow_dispatch" ]]; then
-              if [[ "${{ github.event.ref }}" =~ ^refs/heads/(integration|rel-v([8-9]|[1-9][0-9]+)\.[0-9]+)$ ]]; then
+              if [[ "${{ github.event.ref }}" =~ ^refs/heads/(integration|rel-v([8-9]|[1-9][0-9]+)p[0-9]+)$ ]]; then
```
Comment (fstagni) [inline .github/workflows/deployment.yml:67]: Shouldn't this be

```suggestion
              if [[ "${{ github.event.ref }}" =~ ^refs/heads/(integration|rel-v([8-9]|[1-9][0-9]+)r[0-9]+)$ ]]; then
```
?

**Comment by chaen** [comment]
I think that in general, this whole deployment script would deserve quite some comments to detail the logic. The syntax is quite awful, so it wouldn't hurt :-)

File: .github/workflows/deployment.yml (hunk 1 above)
Comment (chrisburr) [reply .github/workflows/deployment.yml:67]: Yup 🤦

**Comment by chrisburr** [comment]
> I think that in general, this whole deployment script would deserve quite some comments to detail the logic. The syntax is quite awful, so it wouldn't hurt :-)

Done :+1: Though someone other than me should check if [63bb8f0](https://github.com/DIRACGrid/DIRAC/pull/6377/commits/63bb8f0e91c322bae0336b68f6f6016402165a6a) is understandable.

**Comment by fstagni** [comment]
Looks OK to me, I let @chaen to comment if he's happy. 

@chrisburr would you check for probably a similar issue in WebAppDIRAC? https://github.com/DIRACGrid/WebAppDIRAC/blob/integration/.github/workflows/deployment.yml#L69

**Review by fstagni — APPROVED** [review]

**Pull request merged**